### PR TITLE
added admin service, otp functionality and cleaned up

### PR DESCRIPTION
**Pull request by MishaFomenko**
## Describe your changes

- added admin seed service
- fixed types
- added otp verification

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/)

### Backend

- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/auth/dto/response/sign-in.response.dto.ts
Comment (lyudad) [inline src/modules/auth/dto/response/sign-in.response.dto.ts:1]: delete "responce" folder, to deep hierarchy

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -31,15 +31,16 @@ export class AuthController {
 
   @UseGuards(LocalAuthGuard)
   @Post(ERouteName.SIGNIN_ROUTE)
-  @ApiOperation({ summary: AuthControllerDocStrings.operations.SIGNIN.summary })
+  @ApiOperation({ summary: 'Sign in a user' })
   @ApiOkResponse({
-    description: AuthControllerDocStrings.operations.SIGNIN.okResponse,
+    description: 'User signed in successfully',
+    status: 200,
     type: SignUpRequestDto,
```
Comment (eugeneiv1) [inline src/modules/auth/auth.controller.ts:38]: SignUpRequestDto in response??

File: src/modules/auth/dto/sign-in.response.dto.ts [hunk 2]
```diff
@@ -126,24 +126,21 @@ export class UserDto {
...
   @ApiProperty({
     description: 'The access token a user receives to the client',
     example:
       'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwczovL2V4YW1wbGUuYXV0aDAuY29tLyIsImF1ZCI6Imh0dHBzOi8vYXBpLmV4YW1wbGUuY29tL2NhbGFuZGFyL3YxLyIsInN1YiI6InVzcl8xMjMiLCJpYXQiOjE0NTg3ODU3OTYsImV4cCI6MTQ1ODg3MjE5Nn0.CA7eaHjIHz5NxeIJoFK9krqaeZrPLwmMmgI_XiQiIkQ',
   })
   access_token: string;
+
+  @ApiProperty()
+  is_verified: boolean;
 }
```
Comment (eugeneiv1) [inline src/modules/auth/dto/sign-in.response.dto.ts:146]: is it ok to have UserDto, AuthReqDto and AccessResponseDto inside sign-in.response.dto?

File: src/modules/auth/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        ESubjectName.VERIFY,
+        EMailTemplate.VERIFY,
+        { otp_code: otp, name: dto.full_name },
+      ),
+    ]).catch(() => {
+      throw new HttpException(
+        AuthServiceErrors.errors.REGISTER,
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    });
```
Comment (eugeneiv1) [inline src/modules/auth/services/auth.service.ts:66]: Is this catch() needed?

File: src/modules/auth/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    if (user && (await bcrypt.compare(password, user.password))) {
+      return user;
+    }
+    throw new UnauthorizedException(
+      AuthServiceErrors.errors.INVALID_CREDENTIALS,
+    );
+  }
+
+  async login(user: UserDto): Promise<AccesResponseDto> {
```
Comment (eugeneiv1) [inline src/modules/auth/services/auth.service.ts:87]: mb rename login to signIn, so it will correspond to signUp?

**Pull request merged**